### ISSUE TITLE
closeOnSelect not working

### ISSUE THREAD

**Issue by wildhart** [riophae/vue-treeselect#165]
Even on your demo here: https://vue-treeselect.js.org/#more-features the `close-on-select` option does nothing - the menu stays open after selecting an option.

I've managed to implement myself using:
```js
<treeselect ref="searchCat" multiple @select="$refs.searchCat[0].closeMenu()" ...
```

**Comment by sololance** [comment]
Hi!

Looks like it was disabled in this commit https://github.com/riophae/vue-treeselect/commit/6d1370a4c5bf4e03c2221dcf5c635f03d922cfe7 . 

However, looks like documentation mistake, could be used only when selection is single?


@riophae can you take a look please?

Is it possible to enable for **closeOnSelect** for multiple too?

https://github.com/riophae/vue-treeselect/blob/master/src/mixins/treeselectMixin.js#L1814


Thanks!

**Comment by riophae** [comment]
@sololance You're right, `closeOnSelect` only works when selection is single. This is by design and is mentioned in the documentation.

![image](https://user-images.githubusercontent.com/1289884/51920542-96db0400-2420-11e9-8c63-2b5af1721ed5.png)

I'll consider removing this limitation. Thanks for feedback!

**Comment by sololance** [comment]
@riophae well, "use only when `:multiple="true"`" means that  `closeOnSelect` will work only when multiple true, that's the reason why this issue was created 😸 .

I can probably push a PR for that. However, there is a breaking a change `closeOnSelect` default value to `false`. Otherwise `closeOnSelect` will be enabled by default for single and multiple 🤔

**Comment by vietanhbui** [comment]
I use `:closeOnSelect="true"` in mode `:async="true"`, but not working. I think problem is `:async="true"`.
